src/routes/webhooks.ts

```diff
@@ -109,4 +141,36 @@ export default async function (fastify: FastifyInstance): Promise<void> {
       return res.status(204).send();
     },
   });
+
+  fastify.post<{
+    Body: {
+      userId: string;
+      marketingCtaId: string;
+    };
+  }>('/customerio/marketing_cta', {
+    config: {
+      rawBody: true,
+    },
+    handler: async (req, res) => {
+      const valid = verifyCIOSignature(process.env.CIO_WEBHOOK_SECRET, req);
+      if (!valid) {
+        return res.status(403).send({ error: 'Invalid signature' });
+      }
+
+      const { userId, marketingCtaId } = req.body;
+
+      const con = await createOrGetConnection();
+      await con.getRepository(UserMarketingCta).upsert(
+        {
+          userId: userId,
+          marketingCtaId: marketingCtaId,
+        },
+        ['userId', 'marketingCtaId'],
+      );
+
+      await getMarketingCta(con, logger, userId);
```

Comment:
It'll throw a 500 error of either of userId or marketingCtaId doesn't exist.
Should probably wrap it in a try...catch so that we can return a different status

Reply:
Yeah, webhoook providers often disable webhooks that return 2xx status code repediately so I would prefer we catch, log error and return 2xx always.

Reply:
We can return `success: false` in that case as well.

Reply:
👍 I will put it in a try...catch and return false and logs if it fails.

Reply:
I return 400 with `success:false`

Reply:
> We have a 4 second timeout and if we get a 408, 409, 429, or 5xx response code in that period, Customer.io will retry sending the notification several times with exponential backoff. We will retry your request up to 10 times over a period of approximately 3 hours.

Reply:
Nice 👍